billy-core-jpa/src/main/java/com/premiumminds/billy/persistence/entities/jpa/JPAGenericInvoiceEntryEntity.java

```diff
@@ -266,6 +270,11 @@ public AmountType getAmountType() {
         return this.type;
     }
 
+    @Override
+    public Optional<ExternalID<GenericInvoiceEntry>> getExternalID() {
+        return Optional.of(ExternalID.fromValue(externalID));
```

Comment:
`Optional.of` will imply that the field will not actually be optional. Was this on purpose?